yabt/builders/cpp_caching_test.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 Resonai Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+cpp caching tests
+~~~~~~~~~~~~~~~~~~
+
+:author: Dana Shamir
+"""
+
+from os import path
+import os
+from subprocess import check_output
+
+import pytest
+import shutil
+
+from .. import cli
+from ..buildcontext import BuildContext
+from ..graph import populate_targets_graph
+from ..logging import make_logger
+
+PROJECT_DIR = path.join(path.dirname(path.abspath(__file__)), '..', '..',
+                        'tests', 'cpp_caching')
+OP_OBJ_FILE = path.join('yabtwork', 'release_flavor', 'CppLib',
+                        '_binary_operation', 'src', 'binary_operation.o')
+
+
+def build_main_app():
+    basic_conf = cli.init_and_get_conf(
+        ['--non-interactive', '--continue-after-fail', 'build'])
+    build_context = BuildContext(basic_conf)
+    basic_conf.targets = [':main-app']
+    populate_targets_graph(build_context, basic_conf)
+    build_context.build_graph()
+
+
+@pytest.mark.slow
+@pytest.mark.skip(reason="Not supported feature")
+def test_caching(tmp_dir):
+    shutil.copytree(PROJECT_DIR, 'cpp_caching')
+    os.chdir('cpp_caching')
```

Comment:
huh? what's being copied here and where to? this seems fishy

is this because you're modifying the source code as part of the test?

in that case I'd suggest using randomly generated temp dir to run in, so you can safely delete it at the end, and don't need to worry about concurrent test runs

Reply:
Please note the fixture used `tmp_dir`.
The test already runs in a random temporary dir and I copy the files there.

Reply:
oh, missed that, thanks!